test/e2e_new/config/aws-dev.yaml

```diff
@@ -0,0 +1,104 @@
+---
+# E2E test scenario using local dev images and manifests built from the source tree for following providers:
+# - cluster-api
+# - bootstrap kubeadm
+# - control-plane kubeadm
+# - aws
+
+# For creating local dev images built from the source tree;
+# - from the CAPI repository root, `make docker-build REGISTRY=gcr.io/k8s-staging-cluster-api` to build the cluster-api, bootstrap kubeadm, control-plane kubeadm provider images.
+# - from the CAPA repository root, `make docker-build REGISTRY=gcr.io/k8s-staging-cluster-api` to build the aws provider image.
+
+# Following env variables should be set before running the provider
+
+# export AWS_REGION=
+# export AWS_SSH_KEY_NAME=
+# export AWS_B64ENCODED_CREDENTIALS=
+
+images:
+  # Use local dev images built source tree;
+  - name: gcr.io/k8s-staging-cluster-api/cluster-api-controller-amd64:dev
+    loadBehavior: mustLoad
+  - name: gcr.io/k8s-staging-cluster-api/kubeadm-bootstrap-controller-amd64:dev
+    loadBehavior: mustLoad
+  - name: gcr.io/k8s-staging-cluster-api/kubeadm-control-plane-controller-amd64:dev
+    loadBehavior: mustLoad
```

Comment:
Do we actually want to load the images here from the latest upstream, or should we test against a known good release?

Reply:
latest upstream SGTM along with using the latest released version for providers. That will enable devs to get started with the tests with minimal setup.